include/soci/logger.h

```diff
@@ -30,6 +30,11 @@ class SOCI_DECL logger_impl
     // Called to indicate that a new query is about to be executed.
     virtual void start_query(std::string const & query) = 0;
 
+    // Called to log a parameter that is bound to the currently active query
+    virtual void add_query_parameter(std::string name, std::string value) = 0;
+
+    virtual void reset_query_parameter() = 0;
```

Comment:
As this resets _all_ parameters and not just one of them, I'd like to rename it to

```suggestion
    virtual void reset_query_parameters() = 0;
```